test/hotspot/jtreg/vmTestbase/nsk/jdi/ObjectReference/waitingThreads/waitingthreads002.java

```diff
@@ -159,7 +164,7 @@ private static void execTest() {
                     objRef = (ObjectReference) debuggeeClass.getValue(debuggeeClass.fieldByName(fieldName));
                     try {
                         List waitingThreads = objRef.waitingThreads();
-                        if (waitingThreads.size() != waitingthreads002a.threadCount) {
+                        if (waitingThreads.size() != expWaitingCount) {
```

Comment:
I don't see the need for the expWaitingCount bookkeeping. Can't we just verify that size() is zero if we are using virtual threads? I guess maybe the reason you took this approach is because you don't know if the threads are going to be virtual or not until you check them. There is a way to find out, but it's not that pretty either:

    static final boolean vthreadMode = "Virtual".equals(System.getProperty("test.thread.factory"));

Reply:
Thank you for the suggestion. Updated with it.